### PR TITLE
[3544] Add missing positions for inside label

### LINKED ISSUE #3544: Add missing positions for inside label
Currently it is only possible to specify the position TOP_CENTER as position for an insideLabel.
Add support for the other positions : TOP_LEFT, TOP_RIGHT, BOTTOM_CENTER, BOTTOM_LEFT, BOTTOM_RIGHT, MIDDLE_CENTER, MIDDLE_LEFT, MIDDLE_RIGHT

### Expected behavior

image

### PR DESCRIPTION
**Pull request by frouene**
Bug: https://github.com/eclipse-sirius/sirius-web/issues/3544

# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [x] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: packages/diagrams/frontend/sirius-components-diagrams/src/converter/convertDiagram.types.ts [hunk 1]
```diff
@@ -13,13 +13,29 @@
...
 export const AlignmentMap: AlignmentMapType = {
   TOP_CENTER: { primaryAlignment: 'TOP', secondaryAlignment: 'CENTER', isPrimaryVerticalAlignment: true },
+  TOP_LEFT: { primaryAlignment: 'TOP', secondaryAlignment: 'LEFT', isPrimaryVerticalAlignment: true },
+  TOP_RIGHT: { primaryAlignment: 'TOP', secondaryAlignment: 'RIGHT', isPrimaryVerticalAlignment: true },
+  BOTTOM_CENTER: { primaryAlignment: 'BOTTOM', secondaryAlignment: 'CENTER', isPrimaryVerticalAlignment: true },
+  BOTTOM_LEFT: { primaryAlignment: 'BOTTOM', secondaryAlignment: 'LEFT', isPrimaryVerticalAlignment: true },
+  BOTTOM_RIGHT: { primaryAlignment: 'BOTTOM', secondaryAlignment: 'RIGHT', isPrimaryVerticalAlignment: true },
+  MIDDLE_CENTER: { primaryAlignment: 'MIDDLE', secondaryAlignment: 'CENTER', isPrimaryVerticalAlignment: true },
+  MIDDLE_LEFT: { primaryAlignment: 'MIDDLE', secondaryAlignment: 'LEFT', isPrimaryVerticalAlignment: true },
+  MIDDLE_RIGHT: { primaryAlignment: 'MIDDLE', secondaryAlignment: 'RIGHT', isPrimaryVerticalAlignment: true },
```
Comment (pcdavid) [inline packages/diagrams/frontend/sirius-components-diagrams/src/converter/convertDiagram.types.ts:35]: Not directly related to this change, but what's the point of `isPrimaryVerticalAlignment` if it's always true?
Why don't we have `verticalAlignment` and `horizontalAlignment`, which seems simpler and more clear than primary & secondary with an additional boolean to decide how to actually interpret them?

File: packages/diagrams/frontend/sirius-components-diagrams/src/renderer/node/FreeFormNode.tsx [hunk 2]
```diff
@@ -137,9 +142,8 @@ export const FreeFormNode = memo(({ data, id, selected, dragging }: NodeProps<Fr
         onDragOver={onDragOver}
         onDrop={handleOnDrop}
         data-testid={`FreeForm - ${data?.targetObjectLabel}`}>
-        <div style={{ ...backgroundStyle }}>
-          {data.insideLabel && <Label diagramElementId={id} label={data.insideLabel} faded={data.faded} transform="" />}
-        </div>
+        <div style={{ ...backgroundStyle }} />
```
Comment (pcdavid) [inline packages/diagrams/frontend/sirius-components-diagrams/src/renderer/node/FreeFormNode.tsx:146]: Is this normal that this div is now empty?

File: packages/diagrams/frontend/sirius-components-diagrams/src/renderer/node/FreeFormNode.tsx [hunk 3]
```diff
@@ -58,6 +58,11 @@ const imageNodeStyle = (
     height: '100%',
     transform: rotation,
     background: getCSSColor(String(style.background), theme),
+    borderRadius: style.borderRadius,
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    zIndex: -1,
```
Comment (pcdavid) [inline packages/diagrams/frontend/sirius-components-diagrams/src/renderer/node/FreeFormNode.tsx:66]: Is this related to this PR?

**Review by pcdavid — CHANGES_REQUESTED** [review]

File: CHANGELOG.adoc [hunk 4]
```diff
@@ -45,6 +45,7 @@
 - https://github.com/eclipse-sirius/sirius-web/issues/2837[#2837] [gantt] Enhance gantt scroll bar management
 - https://github.com/eclipse-sirius/sirius-web/issues/3441[#3441] [gantt] Use a DateTime widget for AbstractTask entity
 - https://github.com/eclipse-sirius/sirius-web/issues/3539[#3539] [diagram] Add editing context variable in diagram delete tool
+- https://github.com/eclipse-sirius/sirius-web/issues/3544[#3544] [diagram] Add missing positions for inside label
```
Comment (pcdavid) [inline CHANGELOG.adoc:84]: Maybe add a screenshot to illustrate what the new possible positions are?

File: packages/diagrams/frontend/sirius-components-diagrams/src/converter/convertDiagram.types.ts (hunk 1 above)
Comment (frouene) [reply packages/diagrams/frontend/sirius-components-diagrams/src/converter/convertDiagram.types.ts:35]: For this PR, I didn't want to change that part.
There is already an PR on the subject: https://github.com/eclipse-sirius/sirius-web/pull/2833

File: packages/diagrams/frontend/sirius-components-diagrams/src/renderer/node/FreeFormNode.tsx (hunk 2 above)
Comment (frouene) [reply packages/diagrams/frontend/sirius-components-diagrams/src/renderer/node/FreeFormNode.tsx:146]: Yes, this div only displays the background

File: packages/diagrams/frontend/sirius-components-diagrams/src/renderer/node/FreeFormNode.tsx (hunk 3 above)
Comment (frouene) [reply packages/diagrams/frontend/sirius-components-diagrams/src/renderer/node/FreeFormNode.tsx:66]: Yes, the background need to be in absolute position now

File: packages/diagrams/frontend/sirius-components-diagrams/src/converter/convertDiagram.types.ts (hunk 1 above)
Comment (pcdavid) [reply packages/diagrams/frontend/sirius-components-diagrams/src/converter/convertDiagram.types.ts:35]: OK

**Review by pcdavid — APPROVED** [review]
It'd be nice to have a screenshot (we'll probably want one for the blog message), but it's not blocking.

**Comment by frouene** [comment]
> It'd be nice to have a screenshot (we'll probably want one for the blog message), but it's not blocking.

I've added this screenshot to show all the available positions 
insideLabelPositions

File: CHANGELOG.adoc (hunk 4 above)
Comment (frouene) [reply CHANGELOG.adoc:84]: Screenshot added
